src/stcal/outlier_detection/utils.py

```diff
@@ -287,8 +298,39 @@ def calc_gwcs_pixmap(in_wcs, out_wcs, in_shape):
     bb = wcs_bbox_from_shape(in_shape)
     log.debug("Bounding box from data shape: {}".format(bb))
 
-    grid = gwcs.wcstools.grid_from_bounding_box(bb)
-    return np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
+    if not order in [1, 3]:
+        raise ValueError("Interpolation order should be either 1 or 3.")
+
+    if stepsize > 1:
+
+        # Make the 1D grids we want the results on, then make
+        # coarser versions with 10% of the points (minimum 10).
+
+        x = gwcs.wcstools.grid_from_bounding_box(bb[1])
+        y = gwcs.wcstools.grid_from_bounding_box(bb[0])
+
+        x_coarse = np.linspace(x[0], x[-1], max(len(x)//stepsize, 10))
+        y_coarse = np.linspace(y[0], y[-1], max(len(y)//stepsize, 10))
+
+        # Compute the full correction on the sparse grid
+
+        sparsegrid = np.meshgrid(y_coarse, x_coarse)
+
+        pixmap_coarse = reproject(in_wcs, out_wcs)(*sparsegrid)
+
+        # Finally, fill in the rest of the pixels using bilinear
+        # interpolation and return the result.
+
+        fx = interpolate.RectBivariateSpline(x_coarse, y_coarse,
+                                             pixmap_coarse[0], kx=order, ky=order)
+        fy = interpolate.RectBivariateSpline(x_coarse, y_coarse,
+                                             pixmap_coarse[1], kx=order, ky=order)
```

Comment:
`RectBivariateSpline` also takes `bbox` parameter that may be helpful. I am not sure.